server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -399,6 +408,7 @@ public void messageReceived(FieldCapabilitiesNodeRequest request, TransportChann
                     .collect(Collectors.groupingBy(ShardId::getIndexName));
                 final FieldCapabilitiesFetcher fetcher = new FieldCapabilitiesFetcher(indicesService);
                 for (List<ShardId> shardIds : groupedShardIds.values()) {
+                    cancellableTask.ensureNotCancelled();
```

Comment:
I am guessing these individual shardIds iterations are expensive? If the parent was cancelled, this node task would have been cancelled before it started.

Reply:
Shard-level tasks can be expensive, but they can be skipped on 8.2+. This is more important for 7.17 than 8.x.

Reply:
Thanks @benwtrent. I updated this check in https://github.com/elastic/elasticsearch/pull/92051/commits/3145edc456b794c15381d0a898b17f64a4938695.